op-node/rollup/derive/batches.go

```diff
@@ -182,14 +186,36 @@ func checkSpanBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1BlockRe
 		log.Trace("received out-of-order batch for future processing after next batch", "next_timestamp", nextTimestamp)
 		return BatchFuture
 	}
-	if batch.GetTimestamp() < nextTimestamp {
-		log.Warn("dropping batch with old timestamp", "min_timestamp", nextTimestamp)
+	if batch.GetBlockTimestamp(batch.GetBlockCount()-1) < nextTimestamp {
+		log.Warn("span batch has no new blocks after safe head")
 		return BatchDrop
 	}
 
-	// dependent on above timestamp check. If the timestamp is correct, then it must build on top of the safe head.
-	if !batch.CheckParentHash(l2SafeHead.Hash) {
-		log.Warn("ignoring batch with mismatching parent hash", "current_safe_head", l2SafeHead.Hash)
+	// finding parent block of the span batch.
+	// if the span batch does not overlap the current safe chain, parentBLock should be l2SafeHead.
+	parentNum := l2SafeHead.Number
+	parentBlock := l2SafeHead
+	if batch.GetTimestamp() < nextTimestamp {
+		if batch.GetTimestamp() > l2SafeHead.Time {
```

Comment:
Why do we need this check? Below check ` (l2SafeHead.Time-batch.GetTimestamp())%cfg.BlockTime != 0` will cover this case.

Reply:
because timestamps are unsigned.

Reply:
Adding comment would be helpful.